src/main/java/com/taskadapter/redmineapi/internal/RedmineDateUtils.java

```diff
@@ -28,4 +33,75 @@
 	 */
 	public static final LocalDateFormat SHORT_DATE_FORMAT_V2 = new LocalDateFormat(
 			"yyyy-MM-dd");
+        
+        private static final LocalDateFormat[] dateTimeFormats = new LocalDateFormat[] {
+            FULL_DATE_FORMAT,
+            FULL_DATE_FORMAT_V2
+        };
+        
+        // SHORT_DATE_FORMAT_V2 is covered by DatatypeConverter
+        private static final LocalDateFormat[] dateFormats = new LocalDateFormat[] {
+            SHORT_DATE_FORMAT
+        };
+        
+        /**
+         * Try to parse supplied string as a datetime/date and return resulting
+         * date.
+         * 
+         * @param inputString
+         * @return parsed date
+         * @throws IllegalArgumentException in case of parse fail
+         * @throws NullPointerException if inputString is null
+         */
+        public static Date parseDate(String inputString) {
+            // The idea:
```

Comment:
This approach seems dangerous. Why not use several parsers , each for separate occasion. E.g. We know that there is a short date in field1 and long date in field2. We should not mix all possible formats together.
In this case we don't know if we are getting a particular format for a field, but we know it should be long and we can pick one depending on some flag in the field. E.g. "/" at a certain position.